### PR TITLE
Feature/129 csv set delimiter

### LINKED ISSUE #129: Adding Capability to Define Delimiter in csv function
**Is your feature request related to a problem? Please describe.**
The current methodology of the csv function assumes a comma delimiter, and the function allows you to easily move through columns and decode or skip them.  

**Describe the solution you'd like**
Many of the web pulls are using tab or space delimiters, and the csv function would allow for much more efficient and easier decoding if it allowed us to define the delimiter.

**Describe alternatives you've considered**
Current decoding methodologies require me to check at the beginning of each column for missing data before moving forward, either with a new format statement if the column is blank, or with decoding that piece of data.  It results in many more format statements.

### PR DESCRIPTION
**Pull request by MikeNeilson**
## Problem Description

Fixes #129 . 

Describe the problem you are trying to solve.

## Solution

Allow user to set a delimiter. Default is ",".

## how you tested the change

Additional decodes tests were added with the script and input set to the delimiter.

## Where the following done:

- [ ] Tests. Check all that apply:
   - [x] Unit tests created or modified that run during ant test.
   - [ ] Integration tests created or modified that run during integration testing
         (Formerly called regression tests.)
   - [ ] Test procedure descriptions for manual testing
- [x] Was relevant documentation updated?
- [ ] Were relevant config element (e.g. XML data) updated as appropriate

If you aren't sure leave unchecked and we will help guide you to want needs changing where.

### CODE REVIEW

**Comment by MikeNeilson** [comment]
@hoffmanjs, FYI.

File: docs/source/start-decoding.rst [hunk 1]
```diff
@@ -87,11 +87,17 @@ The following table includes the csv parser operation.  This command is
...
-+======================+==========================================================+
-| csv(sens#,...)       | Parse a series of comma separated values                 |
-+----------------------+----------------------------------------------------------+
++-----------------------------+------------------------------------------------------------+
+| **Command**                 | **Description**                                            |
++=============================+============================================================+
+| csv(sens#,...)              | Parse a series of comma separated values                   |
++-----------------------------+------------------------------------------------------------+
+| csv(delimiter=\|,sens#,...) | Parse a series of values separated by a specific delimiter |
+|                             | In this example the pipe, \|, character is used.           |
```
Comment (agilmore2) [inline docs/source/start-decoding.rst:96]: remove commas and add either single or double quotes around the sample delimiter in the text like you do for the whitespace characters below.

File: src/main/java/decodes/decoder/CsvFunction.java [hunk 2]
```diff
@@ -53,12 +54,12 @@ public void execute(DataOperations dd, DecodedMessage decmsg) throws DecoderExce
...
 					break;
 				}
-				if (c == ',')
+				if (dd.checkString(delimiter))
 				{
 					processColumn(col++, sb.toString().trim(), decmsg, dd.getCurrentLine(),
 						fieldStart, dd);
 					sb.setLength(0);
-					fieldStart = dd.getBytePos()+1;
+					fieldStart = dd.getBytePos()+delimiter.length();
```
Comment (agilmore2) [inline src/main/java/decodes/decoder/CsvFunction.java:62]: how does a "delimiter can be more than one character" work here? Are the multiple characters in the delimiter string treated as set of options the delimiter could be, or each field is delimited by all the listed characters?

**Review by agilmore2 — DISMISSED** [review]

File: src/main/java/decodes/decoder/CsvFunction.java (hunk 2 above)
Comment (MikeNeilson) [reply src/main/java/decodes/decoder/CsvFunction.java:62]: Basically you can require that the delimiter be a sequence of characters, or a word. E.g. if you use "bob" as the delimiter all the items are separated by the word "bob". I'll update the docs to make that more clear.

it's not done a lot but I've done it in the past for certain sets of data.

File: KEYS
Comment (ktarbet) [inline KEYS:1]: I'm not sure if adding this KEY is intentional.?

File: src/main/java/decodes/decoder/CsvFunction.java [hunk 3]
```diff
@@ -137,14 +138,30 @@ public void setArguments(String argString, DecodesScript script) throws ScriptFo
 		{
 			String t = st.nextToken();
 			t = t.trim();
-			try
+			if (t.startsWith("delimiter"))
 			{
-				int sensorNum = Integer.parseInt(t);
-				sensorNumbers.add(sensorNum);
+				String[] parts = t.split("=");
+				this.delimiter = parts[1];
```
Comment (ktarbet) [inline src/main/java/decodes/decoder/CsvFunction.java:144]:
```suggestion
				if( parts.length ==2 )
				{
					this.delimiter = parts[1];
				}
```

File: src/test/resources/decodes/db/CsvPipeDelimiterDocSample.input [hunk 4]
```diff
@@ -0,0 +1,4 @@
+ignored header line
+03/01/2014 12:00:00 23.95| 17.2| 8252| 0| 0| 98.1
```
Comment (ktarbet) [inline src/test/resources/decodes/db/CsvPipeDelimiterDocSample.input:2]: why not a pipe delimiter after the time-stamp; before the value?

File: KEYS
Comment (MikeNeilson) [reply KEYS:1]: Well, it was, but shouldn't have been in this branch. I've gone and rebased

File: src/test/resources/decodes/db/CsvPipeDelimiterDocSample.input (hunk 4 above)
Comment (MikeNeilson) [reply src/test/resources/decodes/db/CsvPipeDelimiterDocSample.input:2]: it appears I've simply forgotten it... though now I need to check the csv I copied that from.

Comment (MikeNeilson) [reply src/test/resources/decodes/db/CsvPipeDelimiterDocSample.input:2]: Okay, just checked the decodes script processed the date first and then passes the remaining bits to the CSV, so the CSV function only starts after the date.

File: docs/source/start-decoding.rst (hunk 1 above)
Comment (MikeNeilson) [reply docs/source/start-decoding.rst]: Took me a minute to figure out what was actually wrong but good call.

**Review by ktarbet — APPROVED** [review]

**Pull request merged**